mcs/class/System.XML/Test/System.Xml/XmlSecureResolverTests.cs

```diff
@@ -136,7 +136,7 @@ public void TestAsync ()
 			Assert.AreEqual ("file", resolved.Scheme);
 			var task = sr.GetEntityAsync (resolved, null, typeof (Stream));
 			Assert.That (task.Wait (3000));
-			Assert.IsInstanceOfType (typeof (FileStream), task.Result);
+			Assert.IsTrue (task.Result is Stream);
```

Comment:
why not `is FileStream`? Also can you please include `"Unexpected type: " + task.Result.GetType()` as the third parameter so we know what's wrong when this task fails?